scripts/examples/get_auth_var_signing_certificate.py

```diff
@@ -92,32 +96,157 @@ def process_auth_file(file_path: str) -> bytes:
     if certificate is None:
         raise ValueError("No signing certificate found in the file")
 
-    return certificate
+    sha1_thumbprint = sha1(certificate).digest()
+    sha256_thumbprint = sha256(certificate).digest()
+
+    return certificate, sha1_thumbprint, sha256_thumbprint
+
+
+def get_certificate_info(certificate_der: bytes) -> dict:
+    """Extract certificate information for JSON output."""
+    cert = load_der_x509_certificate(certificate_der)
+
+    # Convert serial number to hexadecimal string (matching Windows certutil format)
+    # The serial number in X.509 certificates is an ASN.1 INTEGER, which is always
+    # stored as a signed value. However, RFC 5280 specifies it should be positive.
+    # Some certificates may have the high bit set, causing Python's cryptography
+    # library to interpret them as negative. We extract the actual bytes and
+    # interpret them as unsigned to match the certificate's actual serial number.
+    serial_number = cert.serial_number
+    if serial_number < 0:
+        # Get the byte length needed to represent the absolute value
+        byte_length = (serial_number.bit_length() + 7) // 8
+        # Convert to unsigned by interpreting as two's complement
+        serial_number = serial_number % (1 << (byte_length * 8))
+
+    serial_hex = format(serial_number, 'x')
+
+    issued_to = cert.subject.rfc4514_string()
+    issued_by = cert.issuer.rfc4514_string()
+
+    return {
+        "serial_number": serial_hex,
+        "issued_to": issued_to,
+        "issued_by": issued_by
+    }
+
+
+def process_single_file(file_path: str, save_der: bool = True) -> None:
+    """Process a single authenticated file and save the certificate."""
+    try:
+        certificate, sha1_thumb, sha256_thumb = process_auth_file(file_path)
+        logging.info("Signing certificate extracted successfully.")
+
+        if save_der:
+            certificate_file = file_path + "_signing_certificate.der"
```

Comment:
does this file_path have an extensions you need to strip?

e.g. if `file_path` is `/path/to/my/file.bin` then this because `path/to/my/file.bin_signing_certificate.der`